core/registry.ts

```diff
@@ -265,15 +228,30 @@ function getItem<T>(
  *     otherwise.
  */
 export function hasItem<T>(type: string | Type<T>, name: string): boolean {
-  type = `${type}`.toLowerCase();
-  name = name.toLowerCase();
+  type = normalizeName(`${type}`);
+  name = normalizeName(name);
   const typeRegistry = typeMap[type];
   if (!typeRegistry) {
     return false;
   }
   return !!typeRegistry[name];
 }
 
+/**
+ * Normalize a string to be used as a key.  By default, lowercase and trimmed.
+ *
+ * @param name The key's name. (Ex. 'FiELd_AnGlE ')
+ * @returns A normalized string. (Ex. 'field_angle')
+ */
+function normalizeName(name: string): string {
+  // Do we need this type checking?  We are using TypeScript after all.
```

Comment:
While we're using TypeScript, a lot of developers aren't, so I've found that it reduces confusion when we check anyway and then issue good warnings. This only applies to developer-facing APIs, not internal-only methods. There are a few places where I wouldn't mind adding more (e.g. an FAQ in the forum is related to leaving off the `workspace` parameter when serializing so if we checked for that case and issued an error we could probably save ourselves some support burden)

That being said, this particular message isn't useful anymore. The original message told you it either had to be a string or had to be a `Blockly.registry.Type` class. So I would probably do that check in `register` that the name is valid, and you can eliminate it here.